fixing lint errors

ssandeep opened the pull request.

ssandeep commented:
fixed the merge conflict

ssandeep commented:
@temaniarpit27 @murthyvitwit plz check as I've used panic in a few places - clerk and staking modules

anilCSE commented on x/clerk/module.go:
```diff
@@ -170,7 +170,9 @@ func (am AppModule) EndBlock(_ sdk.Context, _ abci.RequestEndBlock) []abci.Valid
 func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
 	// TODO : uncomment below line
-	types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
+	if err := types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx)); err != nil {
+		panic(err)
+	}
```
let's avoid panic's. As we gonna have integration tests, we can ignore these errors may be
```suggestion
	_ = types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
```

anilCSE commented on x/staking/client/cli/tx.go:
```diff
@@ -25,7 +25,7 @@ import (
 const ForeignEventName = "Staked"
 
-var logger = helper.Logger.With("module", "staking/client/cli")
+// var logger = helper.Logger.With("module", "staking/client/cli")
```
```suggestion
```

anilCSE commented on x/staking/client/cli/tx.go:
```diff
@@ -85,6 +85,9 @@ func ValidatorJoinTxCmd() *cobra.Command {
 
 			ecdsaPubkey, err := ethcrypto.DecompressPubkey(compressedPubkeyBytes)
+			if err != nil {
+				panic(err)
```
```suggestion
				return err
```

anilCSE commented on x/staking/client/cli/tx.go:
```diff
@@ -172,11 +175,11 @@ func ValidatorJoinTxCmd() *cobra.Command {
-	cmd.MarkFlagRequired(FlagAmount)
-	cmd.MarkFlagRequired(FlagSignerPubkey)
-	cmd.MarkFlagRequired(FlagTxHash)
+	cmd.MarkFlagRequired(FlagBlockNumber)     //nolint
+	cmd.MarkFlagRequired(FlagActivationEpoch) //nolint
+	cmd.MarkFlagRequired(FlagAmount)          //nolint
+	cmd.MarkFlagRequired(FlagSignerPubkey)    //nolint
+	cmd.MarkFlagRequired(FlagTxHash)          //nolint
```
I am ok with this but how about the following? It looks a bit better for me
```suggestion
	_ = cmd.MarkFlagRequired(FlagBlockNumber)
	_ = cmd.MarkFlagRequired(FlagActivationEpoch)
	_ = cmd.MarkFlagRequired(FlagAmount)
	_ = cmd.MarkFlagRequired(FlagSignerPubkey)
	_ = cmd.MarkFlagRequired(FlagTxHash)
```

anilCSE commented on x/staking/client/cli/tx.go:
```diff
@@ -196,6 +199,9 @@ func SignerUpdateTxCmd() *cobra.Command {
 			proposerAddrStr, _ := cmd.Flags().GetString(FlagProposerAddress)
 			proposer, err := sdk.AccAddressFromHex(proposerAddrStr)
+			if err != nil {
+				panic(err)
```
```suggestion
				return err
```

anilCSE commented on x/staking/client/cli/tx.go:
```diff
@@ -216,6 +222,9 @@ func SignerUpdateTxCmd() *cobra.Command {
 
 			ecdsaPubkey, err := ethcrypto.DecompressPubkey(compressedPubkeyBytes)
+			if err != nil {
+				panic(err)
```
```suggestion
				return err
```

anilCSE commented on x/staking/client/cli/tx.go:
```diff
@@ -257,12 +266,12 @@ func SignerUpdateTxCmd() *cobra.Command {
-	cmd.MarkFlagRequired(FlagLogIndex)
-	cmd.MarkFlagRequired(FlagBlockNumber)
-	cmd.MarkFlagRequired(FlagNonce)
+	cmd.MarkFlagRequired(FlagValidatorID)     //nolint
+	cmd.MarkFlagRequired(FlagTxHash)          //nolint
+	cmd.MarkFlagRequired(FlagNewSignerPubkey) //nolint
+	cmd.MarkFlagRequired(FlagLogIndex)        //nolint
+	cmd.MarkFlagRequired(FlagBlockNumber)     //nolint
+	cmd.MarkFlagRequired(FlagNonce)           //nolint
```
ditto as above

anilCSE commented on x/staking/module.go:
```diff
@@ -85,7 +85,9 @@ func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Rout
 func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
 	// TODO : uncomment below line
-	types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
+	if err := types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx)); err != nil {
```
same as above

ssandeep replied:
In this case, for returning the error, I will have to change the function signature which might not be a good idea.
1. Would it be ok to just do
`_ = types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))`
like what we did for `cmd.MarkFlagRequired`
Or
2. panic as this is in module.go
@anilCSE @jdkanani

ssandeep replied:
Implemented (1) as of now. Please let me know if I need to change.

anilCSE reviewed (changes requested):
some nits

jdkanani commented on x/staking/client/cli/tx.go:
```diff
@@ -333,12 +346,12 @@ func StakeUpdateTxCmd() *cobra.Command {
-	cmd.MarkFlagRequired(FlagBlockNumber)
-	cmd.MarkFlagRequired(FlagAmount)
-	cmd.MarkFlagRequired(FlagNonce)
+	cmd.MarkFlagRequired(FlagTxHash)      //nolint
+	cmd.MarkFlagRequired(FlagLogIndex)    //nolint
+	cmd.MarkFlagRequired(FlagValidatorID) //nolint
+	cmd.MarkFlagRequired(FlagBlockNumber) //nolint
+	cmd.MarkFlagRequired(FlagAmount)      //nolint
+	cmd.MarkFlagRequired(FlagNonce)       //nolint
```
We should also ignore err instead of `//nolint`, like -

```
_ = cmd.MarFlagXXX
```

jdkanani commented on x/staking/client/cli/tx.go:
```diff
@@ -282,6 +290,11 @@ func StakeUpdateTxCmd() *cobra.Command {
 			proposerAddrStr, _ := cmd.Flags().GetString(FlagProposerAddress)
 			proposer, err := sdk.AccAddressFromHex(proposerAddrStr)
+
+			if err != nil {
+				panic(err)
+			}
```
Please return err here.

jdkanani commented on x/staking/client/cli/tx.go:
```diff
@@ -257,12 +265,12 @@ func SignerUpdateTxCmd() *cobra.Command {
-	cmd.MarkFlagRequired(FlagLogIndex)
-	cmd.MarkFlagRequired(FlagBlockNumber)
-	cmd.MarkFlagRequired(FlagNonce)
+	cmd.MarkFlagRequired(FlagValidatorID)     //nolint
+	cmd.MarkFlagRequired(FlagTxHash)          //nolint
+	cmd.MarkFlagRequired(FlagNewSignerPubkey) //nolint
+	cmd.MarkFlagRequired(FlagLogIndex)        //nolint
+	cmd.MarkFlagRequired(FlagBlockNumber)     //nolint
+	cmd.MarkFlagRequired(FlagNonce)           //nolint
```
We should also ignore err instead of `//nolint`, like -

```
_ = cmd.MarFlagXXX
```

ssandeep replied:
sure, I am relooking all the places I've made the changes again.

ssandeep commented:
I have removed all the `panic` statements I added and returning `err` where the function signature allows and in other cases, ignoring the error check by assigning to `_`

anilCSE commented on x/clerk/keeper/keeper.go:
```diff
@@ -29,7 +29,7 @@ type (
 		cdc      codec.LegacyAmino
 		storeKey sdk.StoreKey
-		memKey   sdk.StoreKey
+		// memKey   sdk.StoreKey
```
```suggestion
```

anilCSE commented:
@ssandeep can you update your branch with master? Looks like the testscripts are still failing due to that

ssandeep commented:
@anilCSE , you've already merged the master branch right?

anilCSE commented:
> @anilCSE , you've already merged the master branch right?

Ah sorry, yes I did. Testscripts are failing, can you look into them. Should be easy fixes though

ssandeep commented:
Some new linter errors introduced in chainmanager. Closing this for now :D
